add herbstluftwm on linux?

Saw a timeout recently in napari a test that actually manages window focus, when running the test in a different repo that uses this setup instead: https://github.com/pyapp-kit/superqt/actions/runs/12485275920/job/34843848003?pr=268

<details>

<summary>here is the test</summary>

```python
    @skip_local_focus
    @skip_on_mac_ci
    @pytest.mark.key_bindings
    @pytest.mark.parametrize(
        'confirm_key',
        ['enter', 'return', 'tab'],
    )
    def test_preferences_dialog_not_dismissed_by_keybind_confirm(
        qtbot, pref, confirm_key
    ):
        """This test ensures that when confirming a keybinding change, the dialog is not dismissed.
    
        Notes:
            * Skipped on macOS CI due to accessibility permissions not being
              settable on macOS GitHub Actions runners.
            * For this test to pass locally, you need to give the Terminal/iTerm/VSCode
              application accessibility permissions:
                  `System Settings > Privacy & Security > Accessibility`
    
            See https://github.com/asweigart/pyautogui/issues/247 and
            https://github.com/asweigart/pyautogui/issues/247#issuecomment-437668855
        """
        shortcut_widget = (
            pref._stack.widget(3).widget().widget.widgets['shortcuts']
        )
        pref._stack.setCurrentIndex(3)
        # ensure the dialog is showing
        pref.show()
        qtbot.waitExposed(pref)
        assert pref.isVisible()
    
        shortcut = shortcut_widget._table.item(
            0, shortcut_widget._shortcut_col
        ).text()
        assert shortcut == 'U'
    
        x = shortcut_widget._table.columnViewportPosition(
            shortcut_widget._shortcut_col
        )
        y = shortcut_widget._table.rowViewportPosition(0)
    
        item_pos = QPoint(x, y)
        qtbot.mouseClick(
            shortcut_widget._table.viewport(),
            Qt.MouseButton.LeftButton,
            pos=item_pos,
        )
        qtbot.mouseDClick(
            shortcut_widget._table.viewport(),
            Qt.MouseButton.LeftButton,
            pos=item_pos,
        )
>       qtbot.waitUntil(lambda: QApplication.focusWidget() is not None)
E       pytestqt.exceptions.TimeoutError: waitUntil timed out in 5000 milliseconds
```

</details>

the only difference I can see is that their headless setup action https://github.com/aganders3/headless-gui uses [herbstluftwm](https://github.com/aganders3/headless-gui/blob/fbd407c863dc613326c425c219d85507cd886a5f/action.yml#L11-L14) in addition to xvfb.  (as [suggested in pytestqt docs](https://pytest-qt.readthedocs.io/en/latest/troubleshooting.html#xvfb-assertionerror-timeouterror-when-using-waituntil-waitexposed-and-ui-events)).  Would you consider a PR that adds that here?

--- a/README.md
+++ b/README.md
@@ -35,22 +35,30 @@ jobs:
 
 - `qt` (default `false`): set to `true` to install libraries required for Qt
   on Linux, e.g.:
+
   ```yml
       - uses: pyvista/setup-headless-display-action@v3
         with:
           qt: true
   ```
+
 - `pyvista` (default `true`): set to `false` if you don't want to set env
   vars to use PyVista in offscreen mode.
+
+- `window_manager` (default `true`): set to `false` if you don't want to
+  install a window manager (currently herbstluftwm) on Linux.
+  This is only applicable for Linux.
 
 - `mesa3d-release` (default `24.3.0`): set to a specific release to install
   that version of Mesa3D. This is only applicable for Windows. For example,
   to install Mesa3D 21.2.5:
+
   ```yml
       - uses: pyvista/setup-headless-display-action@v3
         with:
           mesa3d-release: 21.2.5
   ```
+
   You can also use `latest` to use the latest release version.
 
 ### 🖼️ PyVista Example

--- a/README.md
+++ b/README.md
@@ -45,9 +45,8 @@ jobs:
 - `pyvista` (default `true`): set to `false` if you don't want to set env
   vars to use PyVista in offscreen mode.
 
-- `window_manager` (default `true`): set to `false` if you don't want to
-  install a window manager (currently herbstluftwm) on Linux.
-  This is only applicable for Linux.
+- `wm` (default `herbstluftwm`): Installs window manager on Linux.
+  set to `false` if you don't want to install a window manager.
 
 - `mesa3d-release` (default `24.3.0`): set to a specific release to install
   that version of Mesa3D. This is only applicable for Windows. For example,

--- a/action.yml
+++ b/action.yml
@@ -36,7 +36,8 @@ runs:
           libglx-mesa0 \
           libgl1 \
           xvfb \
-          x11-xserver-utils
+          x11-xserver-utils \
+          herbstluftwm
 
     - name: Install Linux Qt dependencies
       if: runner.os == 'Linux' && inputs.qt != 'false'
@@ -86,6 +87,13 @@ runs:
         export DISPLAY=:99.0
         echo "DISPLAY=:99.0" >> $GITHUB_ENV
         Xvfb :99 -screen 0 1024x768x24 > /dev/null 2>&1 &
+        sleep 3
+
+    - name: Start window manager on Linux
+      if: runner.os == 'Linux'
+      shell: bash
+      run: |
+        herbstluftwm &
         sleep 3
 
     - name: Configure for PyVista

--- a/action.yml
+++ b/action.yml
@@ -5,11 +5,15 @@ inputs:
   pyvista:
     description: "Set PyVista env vars for headless mode"
     required: false
-    default: true
+    default: "true"
   qt:
     description: "Install libraries required for Qt on Linux"
     required: false
-    default: false
+    default: "false"
+  window_manager:
+    description: "Install herbstluftwm window manager on Linux"
+    required: false
+    default: "true"
   mesa3d-release:
     description: |
       Mesa3D release to install (by default, the latest release is installed).
@@ -90,7 +94,7 @@ runs:
         sleep 3
 
     - name: Start window manager on Linux
-      if: runner.os == 'Linux'
+      if: runner.os == 'Linux' && inputs.window_manager != 'false'
       shell: bash
       run: |
         herbstluftwm &

--- a/action.yml
+++ b/action.yml
@@ -10,10 +10,10 @@ inputs:
     description: "Install libraries required for Qt on Linux"
     required: false
     default: "false"
-  window_manager:
-    description: "Install herbstluftwm window manager on Linux"
+  wm:
+    description: "Install window manager on Linux"
     required: false
-    default: "true"
+    default: "herbstluftwm"
   mesa3d-release:
     description: |
       Mesa3D release to install (by default, the latest release is installed).
@@ -40,8 +40,12 @@ runs:
           libglx-mesa0 \
           libgl1 \
           xvfb \
-          x11-xserver-utils \
-          herbstluftwm
+          x11-xserver-utils
+
+    - name: Install Linux Window manager
+      shell: bash
+      if: runner.os == 'Linux' && inputs.wm == 'herbstluftwm'
+      run: sudo apt-get install -y herbstluftwm
 
     - name: Install Linux Qt dependencies
       if: runner.os == 'Linux' && inputs.qt != 'false'
@@ -94,7 +98,7 @@ runs:
         sleep 3
 
     - name: Start window manager on Linux
-      if: runner.os == 'Linux' && inputs.window_manager != 'false'
+      if: runner.os == 'Linux' && inputs.wm == 'herbstluftwm'
       shell: bash
       run: |
         herbstluftwm &

